Updating README adjusting to GPU support

Carmezim opened the pull request:
Updating current limitations and adding instructions to use GPU on Cmake build.

Carmezim committed:
Update README.md

tensorflow-jenkins commented:
Can one of the admins verify this patch?

thepok commented:
Maybe mention that a ^ has to be added to 
More? -DPYTHON_LIBRARIES=C:/Users/%USERNAME%/AppData/Local/Continuum/Anaconda3/libs/python35.lib
if you want to add more lines.

Carmezim commented:
@thepok May bad. I am doing it now. Thank you!

Carmezim committed:
adding missing "^"

Carmezim committed:
Update

fixing and clarifying instruction on GPU build.

Carmezim committed:
Updating Cmake version due problems with v3.7

Cmake 3.7 is generating problems. 3.5.2 and 3.6 are known good versions. [see here](https://github.com/tensorflow/tensorflow/pull/5071#issuecomment-255501356).

thepok commented:
The last steps in the Readme miss, it took me a long time to figure it out. I did not know what to do with the genereatet PIP file. Maybe add a line describing that you have to do "PIP install genereted_file.whl" or what ever is the real name :)

drpngx commented:
@Carmezim do you want to update the README?

Carmezim commented:
@drpngx I just wanted to make those changes on the Cmake readme to avoid confusing folks regarding GPU support as it is already supported and the readme states otherwise.

drpngx commented:
Jenkins, test this please.

drpngx commented:
What happens after CMake 3.6?

Carmezim commented:
@drpngx when building with v.37 it generates an error referring to a stray character " after "MSVC" in line 81 of `tf_core_gpu_kernels_generated_adjust_contrast_op_gpu.cu.cc.obj.RelWithDebInfo.cmake`, earlier versions build successfully [#5071](https://github.com/tensorflow/tensorflow/pull/5071#issuecomment-255501356).

drpngx commented:
Got it, @mrry and @guschmue might be interested in knowing that CMakeList is broken after 3.6.

Carmezim commented:
@drpngx @guschmue knows as he figured it out with folks testing. Now it seems [#5127](https://github.com/tensorflow/tensorflow/pull/5127) came with something that is generating other errors though that even with v3.6 it is not building successfully.

drpngx commented:
Thanks for identifying the CL!

guschmue commented:
I looked at #5127 and think I have a simple fix for it. Just building a fix and if good I'll send a PR.

guschmue reviewed (approved):
cmake 3.7 does NVCC_OPTIONS correctly. We think it stubles over -D__VERSION__=\"MSVC\". Going to look if we can do something on this but for now its good to ask people to use 3.6 so they don't waste their time.

Carmezim committed:
updating minor detail for coherence.

Carmezim committed:
adding instructions for PIP installation.

 [See #5071](https://github.com/tensorflow/tensorflow/pull/5071#issuecomment-255918807).

Carmezim committed:
fixing typo

zheng-xq commented:
@guschmue, could you look at the latest changes and see if you can LGTM? Thanks.

Carmezim commented on tensorflow/contrib/cmake/README.md:
````diff
@@ -207,6 +202,12 @@ Step-by-step Windows build
+   After installing through PIP using the `.whl` file generated set LD_LIBRARY_PATH variable as:
+   
+   ```
+   set LD_LIBRARY_PATH=D:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\lib\Win32
````
Adding it. Could you please just check if you think the instructions placement is appropriated on Step-by-Step-Build.

Carmezim replied:
Do you rather put the set instruction like:
`set PATH="%PATH%;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\bin;c:\local\cuda\bin`
?

Carmezim committed:
adding gpu environment variables instructions

guschmue reviewed (commented):
LGTM if you fix the PATH mentioned in my comment earlier.

zheng-xq commented:
@guschmue, could you do one final pass before we merge this? Thanks.

guschmue reviewed (commented):
LGTM
